src/utils/supergraph_config.rs

```diff
@@ -98,22 +99,236 @@ pub async fn get_supergraph_config(
     };
 
     // Merge Remote and Local Supergraph Configs
-    let supergraph_config = match (remote_subgraphs, local_supergraph_config) {
-        (Some(remote_subgraphs), Some(local_supergraph_config)) => {
-            let mut merged_supergraph_config = remote_subgraphs.inner().clone();
-            merged_supergraph_config.merge_subgraphs(&local_supergraph_config);
-            let federation_version =
-                resolve_federation_version(federation_version.cloned(), &local_supergraph_config);
-            merged_supergraph_config.set_federation_version(federation_version);
+    let supergraph_config = merge_supergraph_configs(
+        remote_subgraphs.map(|remote_subgraphs| remote_subgraphs.inner().clone()),
+        local_supergraph_config,
+        federation_version,
+    );
+    eprintln!("supergraph config loaded successfully");
+    Ok(supergraph_config)
+}
+
+/// Merge local and remote supergraphs, making sure that the federation version is correct: eg, when
+/// `--graph-ref` is passed, it should be the remote version; otherwise, it should be the local
+/// version
+fn merge_supergraph_configs(
+    remote_config: Option<SupergraphConfig>,
+    local_config: Option<SupergraphConfig>,
+    target_federation_version: Option<&FederationVersion>,
+) -> Option<SupergraphConfig> {
+    match (remote_config, local_config) {
+        (Some(remote_config), Some(local_config)) => {
             eprintln!("merging supergraph schema files");
-            Some(merged_supergraph_config)
+            let mut merged_config = remote_config;
+            merged_config.merge_subgraphs(&local_config);
+            let federation_version =
+                resolve_federation_version(target_federation_version.cloned(), &local_config);
+            merged_config.set_federation_version(federation_version);
+            Some(merged_config)
+        }
+        (Some(remote_config), None) => {
+            let federation_version =
+                resolve_federation_version(target_federation_version.cloned(), &remote_config);
+            let mut merged_config = remote_config;
+            merged_config.set_federation_version(federation_version);
+            Some(merged_config)
         }
-        (Some(remote_subgraphs), None) => Some(remote_subgraphs.inner().clone()),
-        (None, Some(supergraph_config)) => Some(supergraph_config),
+        // WARNING: this ignores the target fed version; should we ignore it?
+        (None, Some(local_config)) => Some(local_config),
```

Comment:
This is an oversight, I feel like the expected behaviour is that if you've given us a local file that has a version in it then we should respect that and not ignore it

Reply:
Yeah, I think there is an oversight here in that `local_config` isn't guaranteed to have a federation_version. This block should do something like:
```
if local_config.get_federation_version().is_none() { 
 local_config.set_federation_version(target_federation_version.unwrap_or(FederationVersion::LatestFedTwo));
}
```